packages/types/src/types.ts

```diff
@@ -218,11 +235,48 @@ export type Transaction = {
   fee: string
 }
 
-export type TxHistoryResponse = {
+type EthereumSpecificFields = {
+  nonce: string
+}
+
+type BitcoinSpecificFields = {
+  opReturn: string
+}
+
+type ChainFieldMap = {
+  [ChainTypes.Ethereum]: EthereumSpecificFields
+  [ChainTypes.Bitcoin]: BitcoinSpecificFields
+}
+
+export type Transaction<T extends ChainTypes = ChainTypes> = TransactionBase<T> & {
+  details: ChainFieldMap[T] & Record<string, undefined>
+}
+
+// unit tests for types
+
+// a generic transaction should have details.nonce of string | undefined
+ta.assert<ta.Equal<Transaction['details']['nonce'], string | undefined>>()
```

Comment:
Is this ta.assert stuff meant to live alongside types of would we want to move out into a .test file just to make this file a bit cleaner to look at?

Reply:
I tried that a while ago on a different project; IIRC it doesn't actually work if we move it out into a test file because of magical jest reasons.

Reply:
Works for me!